nncf/compression_method_api.py

```diff
@@ -106,58 +103,28 @@ def state_dict(self):
         default_keys = {'current_step', '_current_epoch'}
         return {key: val for key, val in self.__dict__.items() if key in default_keys}
 
-    @property
-    def current_epoch(self):
-        return 0 if self._current_epoch == -1 else self._current_epoch
-
-@functools.total_ordering
-class CompressionLevel(Enum):
-    NONE = 0
-    PARTIAL = 1
-    FULL = 2
-
-    # pylint:disable=comparison-with-callable
-    def __add__(self, other: 'CompressionLevel') -> 'CompressionLevel':
-        """
-        Defines compression level of a composite compression controller, consist of two algorithms, where `self` is
-        compression level of first algorithm and other - compression level of second one.
-            NONE    & NONE    = NONE
-            PARTIAL & PARTIAL = PARTIAL
-            FULL    & FULL    = FULL
-            NONE    & PARTIAL = PARTIAL
-            NONE    & FULL    = PARTIAL
-            PARTIAL & FULL    = PARTIAL
-        Args:
-            other: instance of another compression level
-        Returns:
-            common compression level of two algorithms
-        """
-        if self.value == other.value:
-            return self
-        return CompressionLevel.PARTIAL
-
-    def __lt__(self, other: 'CompressionLevel') -> bool:
-        return self.value < other.value
+    def __getattribute__(self, name):
+        if name == 'current_epoch':
+            return 0 if self._current_epoch == -1 else self._current_epoch
```

Comment:
Why didn't `@property` work?

Reply:
Do we really need to return `0` when `_current_epoch == -1`? I think it is a legacy code. Probably it was needed when the method `epoch_step()` was called at the end of an epoch. But now, the scheduler should not do anything while the `epoch_step()` method will not be called. @vshampor Can you please confirm or reject this statement?

Reply:
I, too, think that this code is no longer relevant. Would any of the tests fail if we removed it?

Reply:
Some tests fail if we removed it. I believe, that @andrey-churkin will remove it during work on the implementation of common schedulers.

Reply:
Yes, no problem.